plan/physical_plan_builder.go

```diff
@@ -212,7 +211,10 @@ func (ds *DataSource) convert2PhysicalPlan(prop *requiredProp) (task, error) {
 		}
 	}
 	if !includeTableScan || len(ds.pushedDownConds) > 0 || len(prop.cols) > 0 {
-		for _, idx := range indices {
+		for i, idx := range indices {
+			if !ds.validIndices[i] && len(prop.cols) == 0 {
```

Comment:
If `ds.validIndices` is false and `len(prop.cols)` > 0, can we add a pre-check whether the cols in property don't fit the index at all? We don't need to call `convertToIndexScan` in this case.

Reply:
Yes, I can add a `TODO` here.